interpreter/text/annot.ml

```diff
@@ -22,8 +22,25 @@ module NameMap = Map.Make(struct type t = Ast.name let compare = compare end)
 type map = annot list NameMap.t
 
 let current : map ref = ref NameMap.empty
+let current_source: Buffer.t = Buffer.create 512
 
-let clear () = current := NameMap.empty
+let refill_wrapper inner_refill lexbuf =
```

Comment:
Actually, I think it makes more sense to move this logic to `Parse`, so that `Annot` doesn't need to know about lexbufs. Here I'd just leave a bare `reset` function that clears name map and source.